[20248] Display errors in the WP details pane via NotificationsService

floriank opened the pull request:
This PR is a recreation of the original PR (which went against `release/4.3` which is obsolete now).

The original PR can be found at #3279 .

Original PR Text:

---

This will change the current errors (which are attached to the the individual field at the moment) into a notification provided by the previously introduced `NotificationsService`.

This Pr will thereby meet the requirements for https://community.openproject.org/work_packages/20248.
## Things left to do/discuss
- [x] Accessibility - I think the plan from the specification might or might not work (currently, the focus stays with the erroneous field
- [x] "continue editing" - see above

Addendum: both discussions have been resolved in the original PR.
## Out of scope
- "Create another WP like this" functionality in successful updates and/or after successful creation

Commit:
intitial display of error messages via the notification service

Signed-off-by: Florian Kraft <f.kraft@finn.de>

Commit:
sneak in a fix for the attachments

this will corect an error thrown when a validaiton error happens -
the form will, in certain edge cases try to submit the work package files
which can be empty at times.

Signed-off-by: Florian Kraft <f.kraft@finn.de>

Commit:
show the errors when the requests have completed

Signed-off-by: Florian Kraft <f.kraft@finn.de>

Commit:
actually use the notifciations in protractor tests

Signed-off-by: Florian Kraft <f.kraft@finn.de>

Commit:
correct test for new error notification

Signed-off-by: Florian Kraft <f.kraft@finn.de>

Commit:
remove obsolete error tests

Signed-off-by: Florian Kraft <f.kraft@finn.de>

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://floriank-feature-20248-error-collection.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://floriank-feature-20248-error-collection.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

Commit:
only show errors when there are errors to be shown

Errors were alwas shown by accident. As the EditableFieldsState object is not
cleared upon successful request, using `finally` would show the error once more
for a successful api call.

Signed-off-by: Florian Kraft <f.kraft@finn.de>

floriank commented:
The glitch found by @ulferts in #3279 has been fixed. 

I'd say wait for the test run and merge.

ulferts commented:
The glitch is fixed indeed and I will merge this now. Please note, that IMO  https://community.openproject.org/work_packages/20248 should cover the success messages, too. I'll write it down on OP.org.
